src/compressor/Compressor.h

```diff
@@ -40,8 +41,21 @@ class Compressor {
 #ifdef HAVE_BROTLI
     COMP_ALG_BROTLI = 5,
 #endif
-    COMP_ALG_LAST	//the last value for range checks
+    COMP_ALG_LAST   //the last value for range checks
+  };
+
+  inline static const std::map<const std::string, const CompressionAlgorithm> compression_algorithms {
```

Comment:
i opened https://github.com/ceph/ceph/pull/21801 to make this a constexpr initializer_list

Reply:
this exposes `compression_algorithms` directly as an interface to determine which are enabled. consider extending the static functions to do this instead. for example:
```c++
  static constexpr std::optional<const char*> get_comp_alg_name(int a);
  static constexpr std::optional<CompressionAlgorithm> get_comp_alg_type(std::string_view s);
```
the caller can loop over `[0,COMP_ALG_LAST)` and call `get_comp_alg_name(i)` to include the names of each algorithm enabled. these functions can be constexpr thanks to initializer_list and the constexpr constructors of std::optional and std::string_view. std::find_if() isn't constexpr until c++20, but a ranged-for loop would work instead

(this would also address the comment `return "???"; // It would be nice to revise this...`)

Reply:
Yeah, 'return "???"' was what the prior code did, and frankly I'm not a fan of that. I decided to lift up only a single level of abstraction because I didn't know how much other code might be using it, but there's certainly no reason it can't be revisited! Thanks for your thoughtful comments.